app/assets/javascripts/mobile/mobile_drawer_swipe.js

```diff
@@ -0,0 +1,136 @@
+(function(){
+  Diaspora.Mobile.Drawer = {
+    initialize: function(){
+      this.drawer = $("#drawer");
+      this.mainContent = $("#main");
+      this.dragZone = this.createDragZone();
+      this.mainContent.append(this.dragZone);
+
+      var self = this;
+      var acc = 0;
+      var hammerOpts = {
+        "prevent_default": false,
+        threshold: 0,
+        direction: Hammer.DIRECTION_HORIZONTAL
+      };
+
+      $("#menu-badge").on("tap click", function(evt){
+        evt.preventDefault();
+        self.createOverlay();
+        self.openDrawer();
+      });
+
+      this.mainContent.on("tap click", function(){
+        if(self.isDrawerOpened()){
+          self.closeDrawer();
+        }
+      });
+
+      this.dragZone.hammer(hammerOpts).bind("panstart", function(){
+          self.createOverlay();
+          $("body").css("overflow", "hidden");
+          self.drawer.addClass("gesture").removeClass("closed-drawer").removeClass("opened-drawer");
+        })
+        .bind("pan", function(e){
+          // Optimization : ignore 2 event over 3
```

Comment:
Correct debouncing does the following (psuedo code incoming):

if (Date.now() - date_last_event > debouncing_threshold)
  return e.preventDefault();
date_last_event = Date.now();